.github/workflows/build-test.yml

```diff
@@ -280,55 +280,72 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        jupyter: [raw, local]
-        python: [nonConda, conda]
-        pythonVersion: ['3.9', '3.10']
-        webOrDesktop: ['']
+        jupyterConnection: [raw] # valid values include raw = uzing zqm, local = using local jupyter, remote = using remote jupyter, web = using remote jupyter in web mode
+        python: [python]  # valid values include python,conda,noPython
+        pythonVersion: ['3.10']
         # Whether we're using stable (<empty>) or pre-release versions of Python packages.
         # When installing pre-release versions, we're only focused on jupyter & related packages.
         # Not pre-release versions of pandas, numpy or other such packages that are not core to Jupyter.
         packageVersion: ['']
+        tags: ['^[^@]+$|@mandatory', '@kernelCore', '@widgets', '@iw', '@webview|@export|@lsp|@variableViewer', '@debugger', '@kernelPicker|@python|@jupyter', '@kernelInstall']
         # We're not running CI on macOS for now because it's one less matrix entry to lower the number of runners used,
         # macOS runners are expensive, and we assume that Ubuntu is enough to cover the UNIX case.
         os: [ubuntu-latest]
         # More details on includes/excludes can be found here https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-additional-values-into-combinations
         # Basically with exclude, you can exclude any of the combinations from the result matrix.
         # & with include, you can include additional items to the result matrix.
         include:
-          - jupyter: remote # We don't care how remote is started (we only want to test connecting to it)
-            python: nonConda
-            pythonVersion: '3.9'
-            webOrDesktop: 'desktop'
-            os: ubuntu-latest
-          - jupyter: remote
-            python: nonConda
-            pythonVersion: '3.9'
-            webOrDesktop: 'web'
+        # Mandatory tests
+          - jupyterConnection: remote
+            python: python
+            pythonVersion: '3.10'
             os: ubuntu-latest
-          - jupyter: raw
-            python: noPython
+            tags: '^[^@]+$|@mandatory'
```

Comment:
is having a regex in the tag necessary?

Reply:
this is a regex that micha supports
mocha only supports regexes (is documented), i don't want to write custom code to filter tests